dashboard/app/models/concerns/serialized_properties.rb

```diff
@@ -19,8 +19,12 @@ def assign_attributes(new_attributes)
     new_properties = attributes.delete('properties').try(:stringify_keys)
 
     super(attributes)
-    # If the properties hash is explicitly assigned then merge its keys with existing properties
-    # instead of replacing the entire hash
+
+    # If the properties hash is explicitly assigned, merge its keys with
+    # existing properties instead of replacing the entire hash. Make sure to
+    # invoke `init_properties` again first, in case the previous `super` call
+    # reset the attributes cache (see refresh_activemodel_attributes_cache.rb).
+    init_properties
```

Comment:
This feels a bit janky, but I'm not sure if it's janky enough to justify spending more time making it less so. Let me know if you think it is and I'll be happy to do some more digging